22-10-2020/main.js

```diff
@@ -0,0 +1,187 @@
+//     { id: 1, name: 'Nguyễn Thị Tèo', score: 9.2 },
+//     { id: 2, name: 'Phạm Văn Bưởi', score: 2.3 },
+//     { id: 3, name: 'Hoàng Văn Nam', score: 3.7 },
+//     { id: 4, name: 'Vũ Ngọc Duy', score: 6.9 },
+//     { id: 5, name: 'Nguyễn Minh Nhật', score: 5.2 },
+//     { id: 6, name: 'Phí Duy Quân', score: 9.6 },
+//     { id: 7, name: 'Trần Minh Minh', score: 6.1 }
+// ]
+const studentNames = [
+    { id: 1, name: 'Nguyễn Thị Tèo' },
+    { id: 2, name: 'Phạm Văn Bưởi' },
+    { id: 3, name: 'Hoàng Văn Nam' },
+    { id: 4, name: 'Vũ Ngọc Duy' },
+    { id: 5, name: 'Nguyễn Minh Nhật' },
+    { id: 6, name: 'Phí Duy Quân' },
+    { id: 7, name: 'Trần Minh Minh' }
+]
+const studentScores = [
+    { id: 1, score: 9.2 },
+    { id: 2, score: 2.3 },
+    { id: 3, score: 3.7 },
+    { id: 4, score: 6.9 },
+    { id: 5, score: 5.2 },
+    { id: 6, score: 9.6 },
+    { id: 7, score: 6.1 }
+]
+const getStudentsArray = (studentNames, studentScores) => {
+    for (let i = 0; i < studentNames.length; i++) {
+        for (let j = 0; j < studentScores.length; j++) {
+            if (studentNames[i].id === studentScores[j].id) {
+                studentNames[i].score = studentScores[j].score
+            }
+        }
+    }
+    return studentNames
+}
+
+// Bài 8
+// Cho array dưới đây là danh sách các students
+// const students = [
+//     { id: 1, name: 'Nguyễn Thị Tèo', score: 9.2 },
+//     { id: 2, name: 'Phạm Văn Bưởi', score: 2.3 },
+//     { id: 3, name: 'Hoàng Văn Nam', score: 3.7 },
+//     { id: 4, name: 'Vũ Ngọc Duy', score: 6.9 },
+//     { id: 5, name: 'Nguyễn Minh Nhật', score: 5.2 },
+//     { id: 6, name: 'Phí Duy Quân', score: 9.6 },
+//     { id: 7, name: 'Trần Minh Minh', score: 6.1 }
+// ]
+// Hãy tìm sinh viên có điểm (score) thấp nhất và sinh viên có score cao nhất
+// Với format đầu ra dạng:
+// return {
+//     theBest: { id: 6, name: 'Phí Duy Quân', score: 9.6 },
+//     theBad: { id: 2, name: 'Phạm Văn Bưởi', score: 2.3 }
+// }
+const students = [
+    { id: 1, name: 'Nguyễn Thị Tèo', score: 9.2 },
+    { id: 2, name: 'Phạm Văn Bưởi', score: 2.3 },
+    { id: 3, name: 'Hoàng Văn Nam', score: 3.7 },
+    { id: 4, name: 'Vũ Ngọc Duy', score: 6.9 },
+    { id: 5, name: 'Nguyễn Minh Nhật', score: 5.2 },
+    { id: 6, name: 'Phí Duy Quân', score: 9.6 },
+    { id: 7, name: 'Trần Minh Minh', score: 6.1 }
+]
+const findMinMaxScore = array => {
+    let theBest = []
+    let theBad = []
+    let maxScore = students[0].score
+    let minScore = students[0].score
+    for (let i = 1; i < students.length; i++) {
+        if (minScore > students[i].score) {
+            minScore = students[i].score
+        }
+        if (maxScore < students[i].score) {
+            maxScore = students[i].score
+        }
+    }
+
+    for (let i = 1; i < students.length; i++) {
+        if (students[i].score == min) {
+            theBad.id = students[i].id
+            theBad.name = students[i].name
+            theBad.score = students[i].score
```

Comment:
```suggestion
            theBad.push(students[i])
```
Bên trên gán `theBad` là `Array` bên dưới lại dùng dấu `.` (cách dùng của obj) ---> viết code nhưng không hề chạy thử